### PR TITLE
fix: require java_outer_classname, not allow java_multiple_files, #239

### LINKED ISSUE #239: Compilation problems with Actions without java_outer_classname
Action codegen generates invalid code if there isn't a `java_outer_classname` option. That's not the case of the other components. 

See https://github.com/lightbend/akkaserverless-java-sdk/pull/237/files#diff-e2cacc5b119d9b80e7513c32929fe8e3603b078893d30718ea82c7024482db8dR19

### PR DESCRIPTION
**Pull request by patriknw**
* at least for now we don't support java_multiple_files=true and java_outer_classname is required
* better with fail fast with clear error message than weird compilation errors

Refs #239

### CODE REVIEW

**Review by octonato — APPROVED** [review]

**Review by johanandren — APPROVED** [review]

File: codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala [hunk 1]
```diff
@@ -83,13 +85,16 @@ class ModelBuilderSuite extends munit.FunSuite {
...
         PackageNaming(
+          "Empty",
           "Empty",
           "google.protobuf",
           Some("google.golang.org/protobuf/types/known/emptypb"),
           Some("com.google.protobuf"),
           Some("EmptyProto"),
           javaMultipleFiles = true)
 
+      println(s"# googleEmptyProto $googleEmptyProto") // FIXME
```
Comment (raboof) [inline codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala:96]: to be removed?

File: codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/SourceGenerator.scala [hunk 2]
```diff
@@ -134,6 +138,26 @@ object SourceGenerator {
...
   }
 
+  private def validate(model: ModelBuilder.Model): Unit = {
+    // at least for now we don't support java_multiple_files=true and java_outer_classname is required
+    def validate(pkgNaming: PackageNaming): List[String] = {
+      var errors = List.empty[String]
+      if (pkgNaming.javaOuterClassnameOption.isEmpty)
+        errors = s"${pkgNaming.protoFileName} must define java_outer_classname" :: errors
+      if (pkgNaming.javaMultipleFiles)
+        errors = s"${pkgNaming.protoFileName} must not define java_multiple_files" :: errors
```
Comment (raboof) [inline codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/SourceGenerator.scala:148]:
```suggestion
        errors = s"${pkgNaming.protoFileName} must not enable java_multiple_files" :: errors
```

**Review by raboof — APPROVED** [review]
Good to be explicit

File: codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala [hunk 3]
```diff
@@ -83,13 +85,16 @@ class ModelBuilderSuite extends munit.FunSuite {
 
       val googleEmptyProto =
         PackageNaming(
+          "Empty",
```
Comment (pvlugter) [inline codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala:88]: Not important for the tests, but could be better to give an `"empty.proto"` filename?

File: codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala [hunk 4]
```diff
@@ -83,13 +85,16 @@ class ModelBuilderSuite extends munit.FunSuite {
...
           "Empty",
           "google.protobuf",
           Some("google.golang.org/protobuf/types/known/emptypb"),
           Some("com.google.protobuf"),
           Some("EmptyProto"),
           javaMultipleFiles = true)
 
+      println(s"# googleEmptyProto $googleEmptyProto") // FIXME
+
```
Comment (pvlugter) [inline codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala:97]:
```suggestion
```

File: codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/SourceGenerator.scala (hunk 2 above)
Comment (pvlugter) [reply codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/SourceGenerator.scala:148]: 👍🏼 `enable` looks good

**Review by pvlugter — APPROVED** [review]

**Comment by pvlugter** [comment]
I'll merge this and then take a look at how far off it is from supporting the option variations properly.

File: codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala (hunk 3 above)
Comment (patriknw) [reply codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala:88]: That was what I started with, but then there were many test failures, so I changed to what the tests were expecting. I think there is a special case for the Empty.

**Pull request merged**